tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/cli/ContentGenerator.java

```diff
@@ -35,5 +43,40 @@
     })
 public abstract class ContentGenerator<API extends NessieApiV2> {
 
-  public abstract API createNessieApiInstance();
+  @CommandLine.Option(
+      names = {"-u", "--uri"},
+      scope = CommandLine.ScopeType.INHERIT,
+      description = "Nessie API endpoint URI, defaults to http://localhost:19120/api/v2.")
+  private URI uri = URI.create("http://localhost:19120/api/v2");
+
+  @CommandLine.Option(
+      names = "--nessie-client",
+      description = "Name of the Nessie client to use, defaults to HTTP suitable for REST.")
+  private String nessieClientName;
+
+  @CommandLine.Option(
+      names = "--nessie-option",
```

Comment:
nit: maybe add `-O` for manual typing convenience?

Reply:
doen